vnext/Microsoft.ReactNative/HStringHelper.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <pch.h>
+
+#include "HStringHelper.h"
+#include "Unicode.h"
+
+#include <winrt/base.h>
+
+namespace winrt {
+
+std::string HSTRINGToString(HSTRING hstring) {
+  std::wstring wstr = WindowsGetStringRawBuffer(hstring, nullptr /*length*/);
```

Comment:
>WindowsGetStringRawBuffer [](start = 22, length = 25)

I believe hstrings can contain embedded nulls which would result in this line behaving improperly